### ISSUE TITLE
Clarify what the "context" claim provides for OIDC tokens

### ISSUE THREAD

**Issue by TomChapple** [github/docs#34572]
### Code of Conduct

- [X] I have read and agree to the GitHub Docs project's [Code of Conduct](https://github.com/github/docs/blob/main/.github/CODE_OF_CONDUCT.md)

### What article on docs.github.com is affected?

https://docs.github.com/en/actions/security-for-github-actions/security-hardening-your-deployments/about-security-hardening-with-openid-connect

### What part(s) of the article would you like to see updated?

In this article, the concept of claim keys is introduced including how to specify them for an organisation and/or a repository. This is particularly helpful with the example OIDC token to relate what values would be available if a particular claim was requested.

However, the "context" claim (used in the default claim template) does not have a corresponding entry in the OIDC token. It makes sense that is doesn't, but it also leaves this claim as very ambiguous to its content and purpose, especially when it is excluded from the list of GitHub-provided claims.

Some sections refer to "context" and show how to use it along with some examples of its outputs listed in the "Example subject claims" (albeit without any mention of "context"). These examples do not show an obvious pattern in formatting, often leading me to questions such as "which order of 'ref' and 'environment' should I expect?". This uncertainty leads me towards explicitly specifying the individual claims instead of using the default, even if it could conceptually meet my requirements.

I understand that the "context" claim (from experimentation) changes its form depending on how the workflow was triggered. Although this might be inferred, there is no explicit answer to inform the user, reducing the confidence in using this claim.

I believe some description of the "context" claim as though it were a separate claim would be very beneficial. Given its dynamic state, details of what it includes and when it includes it would be very useful in determining whether the client requires additional claims. In the end, I want to be confident in knowing that this claim provides what I require.

### Additional information

This was originally explored in Issue #22283, but was closed after addressing one of its other points on accessing the "repository" claim via "repo". Its other point regarding the ambiguity of "context" may not have been addressed.

**Comment by nguyenalex836** [comment]
@TomChapple Thank you for raising this issue! I'll get this triaged for review :sparkles: Our team will provide feedback regarding the best next steps for this issue - thanks for your patience! 💛

**Comment by nguyenalex836** [comment]
@TomChapple Thank you for your patience while our SME team reviewed! 💛 Our SMEs responded with the following -

> `context` is not a claim, but rather a way to provide additional instruction to the external resource provider when consuming the token such as branch or environment. See [example](https://docs.github.com/en/actions/security-for-github-actions/security-hardening-your-deployments/about-security-hardening-with-openid-connect#example-subject-claims) and this [doc](https://docs.github.com/en/actions/security-for-github-actions/security-hardening-your-deployments/about-security-hardening-with-openid-connect#example-requiring-a-reusable-workflow-and-other-claims).

I'll go ahead and close this issue 💛 Thank you for your willingness to contribute to our docs! 🙇

**Comment by TomChapple** [comment]
I understand that "context" is not a claim in the same sense as other claims like "environment" and "job_workflow_ref" (i.e. those listed in [the example OIDC token](https://docs.github.com/en/actions/security-for-github-actions/security-hardening-your-deployments/about-security-hardening-with-openid-connect#understanding-the-oidc-token)). It does indeed provide information in the token's subject that an external resource provider can use, but there is a lack of detail on what this involves.

The closest information we have are the examples that were linked by the SME, but that is all they are described as: examples. Those examples don't mention the "context" claim at all in that section, with the only link being the "doc" link provided by the SME that happens to also link to these examples:

> This example also demonstrates how to use `"context"` to define your conditions. This is the part that follows the repository in the [default `sub` format](https://docs.github.com/en/actions/deployment/security-hardening-your-deployments/about-security-hardening-with-openid-connect#example-subject-claims). For example, when the job references an environment, the context contains: `environment:ENVIRONMENT-NAME`.

I see this as a very loose thread to join the two together and, without experimentation, requires making assumptions on what it can or can't accomplish. I was asking myself "what exactly will it provide in my particular circumstance" and read this as what it _can_ provide instead.

I feel like what would have helped my understanding would involve:

* Identifying that `"context"` is a special case: it does not produce a key with the same name, but instead produces other keys.
* A description of rules on when and what it will contain depending on the triggered workflow that requests an OIDC token. This might even be able to tag along with the "Example subject claims" section if they cover those rules.

In summary, I now understand the function of `"context"`, but I felt as though it was not very clear as to its contribution and how it can be reliably used in workflows. That is what I feel needs to be addressed in the documentation.

**Comment by Lowprofile666-hub** [comment]
CDL

**Comment by TomChapple** [comment]
Is there any further comment on the above point? Although it provides a bit of functional context, I'm not sure the explanation given by the SME addresses what I believe is a gap in the documentation of these OIDC claims.

**Comment by nguyenalex836** [comment]
@TomChapple Thank you for the nudge here! We had previously asked our SMEs if they could review your latest comment - I'll give them another nudge on this as well 💛

**Comment by Steve-Glass** [comment]
>In summary, I now understand the function of "context", but I felt as though it was not very clear as to its contribution and how it can be reliably used in workflows. That is what I feel needs to be addressed in the documentation.

My apologies for the delay and glad to hear you are no longer blocked! I agree with your assessment. We have some updates coming for OIDC in the next few months. My plan is to include this clarification with the other OIDC docs updates we have planned.
